src/runtime_src/core/pcie/linux/pcidrv.cpp

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2022 Advanced Micro Devices, Inc. All rights reserved.
+
+#include <filesystem>
+#include "pcidrv.h"
+
+namespace xrt_core { namespace pci {
+
+void
+drv::
+scan_devices(std::vector<std::shared_ptr<dev>>& ready_list,
+             std::vector<std::shared_ptr<dev>>& nonready_list) const
+{
+  const std::string drv_root = "/sys/bus/pci/drivers/";
+  const std::filesystem::path drvpath = drv_root + name();
+
+  if (!std::filesystem::exists(drvpath))
+    return;
+
+  // Gather all sysfs directory and sort
+  std::vector<std::filesystem::path> vec{
+    std::filesystem::directory_iterator(drvpath),
+    std::filesystem::directory_iterator()
+  };
+  std::sort(vec.begin(), vec.end());
+
+  for (auto& path : vec) {
+    try {
+      auto pf = create_pcidev(path.filename().string());
+
+      // In docker, all host sysfs nodes are available. So, we need to check
+      // devnode to make sure the device is really assigned to docker.
+      if (!std::filesystem::exists(pf->get_subdev_path("", -1)))
+        continue;
+
+      // Insert detected device into proper list.
+      if (pf->is_ready)
+        ready_list.push_back(pf);
+      else
+        nonready_list.push_back(pf);
+    } catch (const std::invalid_argument& ex) {
```

Comment:
```suggestion
    } 
    catch (const std::invalid_argument& ex) {
```

Reply:
Oh, 'catch' statement should start a new line? OK.

Reply:
Just my preference in general to be at the left as much as possible.
```
if (...) {
  stmt1;
  stmt2;
}
else {
  stmt3;
  stmt4
}
```

```
try {
  ...;
}
catch (...) {
}
```

Reply:
Sure.